hash/hash_comparator.go

```diff
@@ -106,11 +109,35 @@ func ComparePbkdf2(_ context.Context, password []byte, hash []byte) error {
 	return errors.WithStack(ErrMismatchedHashAndPassword)
 }
 
+func CompareScrypt(_ context.Context, password []byte, hash []byte) error {
+	// Extract the parameters, salt and derived key from the encoded password
+	// hash.
+	p, salt, hash, err := decodeScryptHash(string(hash))
+	if err != nil {
+		return err
+	}
+
+	// Derive the key from the other password using the same parameters.
+	otherHash, err := scrypt.Key(password, salt, int(p.Cost), int(p.Block), int(p.Parrellization), int(p.KeyLength))
+	if err != nil {
+		return err
```

Comment:
I think you could force this error by supplying a cost that is not a power of 2, e.g. 5. Just so the codecov is happy. :)

Reply:
Sounds good, I did this [here](https://github.com/ory/kratos/pull/2689/files#diff-275881c2b9629b0999bee794700468eea536ffa1e5b2078beb854816435b4c30R243), I still see the codecov error though, but if I run the test I get an ok

Reply:
Done [here](https://github.com/ory/kratos/pull/2689/files#diff-275881c2b9629b0999bee794700468eea536ffa1e5b2078beb854816435b4c30R242)